base/networking.c

```diff
@@ -762,3 +762,341 @@ ipv6_is_enabled ()
+   * Destination and but it into the to be returned list of routes.*/
+  while (1)
+    {
+      gchar *interface;
+      unsigned long mask;
+      unsigned long dest;
+      gchar *char_p;
+      int count = 0;
+
+      /* Get new line. */
+      status = g_io_channel_read_line (file_channel, &line, NULL, NULL, &err);
+      if ((status != G_IO_STATUS_NORMAL) || !line || err)
+        {
+          g_warning (
+            "%s: %s", __func__,
+            err ? err->message
+                : "g_io_channel_read_line() status != G_IO_STATUS_NORMAL");
+          err = NULL;
+          if (line)
+            g_free (line);
+          break;
+        }
+
+      /* Get items in line. */
+      items_in_line = g_strsplit (line, "\t", -1);
+      /* Check for missing entries in line of "/proc/net/route". */
+      for (; items_in_line[count]; count++)
+        ;
+      if (11 != count)
+        {
+          g_strfreev (items_in_line);
+          g_free (line);
+          continue;
+        }
+
+      interface = g_strndup (items_in_line[0], 64);
+      /* Cut interface str after ":" if IP aliasing is used. */
+      if ((char_p = strchr (interface, ':')))
+        {
+          *char_p = '\0';
+        }
+      dest = strtoul (items_in_line[1], NULL, 16);
+      mask = strtoul (items_in_line[7], NULL, 16);
+
+      /* Fill GSList entry. */
+      entry = g_malloc0 (sizeof (struct route_entry));
+      entry->interface = interface;
+      entry->dest = dest;
+      entry->mask = mask;
+      routes = g_slist_append (routes, entry);
+
+      g_strfreev (items_in_line);
+      if (line)
+        g_free (line);
+    }
+
+  status = g_io_channel_shutdown (file_channel, TRUE, &err);
+  if ((G_IO_STATUS_NORMAL != status) || err)
+    g_warning ("%s: %s", __func__,
+               err ? err->message
+                   : "g_io_channel_shutdown() was not successfull");
+
+  return routes;
+}
+
+/**
+ * @brief Get Interface which should be used for routing to destination addr.
+ *
+ * This function should be used sparingly as it parses /proc/net/route for
+ * every call.
+ *
+ * @param[in]   storage_dest    Destination address.
+ * @param[out]  storage_source  Source address. Is set to either address of the
+ * interface we use or global source address if set. Only gets filled if
+ * storage_source != NULL.
+ *
+ * @return Interface name of interface used for routing to destination address.
+ * NULL if no interface found or Error.
+ */
+gchar *
+gvm_routethrough (struct sockaddr_storage *storage_dest,
+                  struct sockaddr_storage *storage_source)
+{
+  if (!storage_dest)
+    return NULL;
+
+  struct ifaddrs *ifaddr = NULL;
+  struct ifaddrs *ifa = NULL;
+  gchar *interface_out = NULL;
+
+  if (getifaddrs (&ifaddr) == -1)
+    {
+      g_debug ("%s: getifaddr failed: %s", __func__, strerror (errno));
+      return NULL;
+    }
+
+  /* IPv4. */
+  if (storage_dest->ss_family == AF_INET)
+    {
+      GSList *routes = NULL;
```

Comment:
This is redundant because it's defined directly below.  Recommend doing the init's after the declaration, with a line separating them, to avoid this kind of confusion.